### DISCUSSION TITLE
How to set upload timeout when server closed?

### DISCUSSION THREAD

**Discussion by mengqingmeng** [aws/aws-sdk-cpp#2263] [Q&A] [answered]
Putting object to server will request about 60s - 70s，when my minio server is cloes. After 60-70 seconds PutObject api will return false result.
My question is: how to reduce the time of request? blow is my client config code:
``` 
// client config 
Aws::Client::ClientConfiguration cfg;
cfg.endpointOverride = endpoint;  // S3服务器地址和端口
cfg.scheme = Aws::Http::Scheme::HTTP;
cfg.verifySSL = false;
cfg.connectTimeoutMs = 3000;
cfg.httpRequestTimeoutMs = 3000;
cfg.requestTimeoutMs = 3000;
cfg.tcpKeepAliveIntervalMs = 15000;
client_ = std::make_shared<Aws::S3::S3Client>(Aws::Auth::AWSCredentials(accessKey, secretKey), cfg,
    Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy::Never, false);
```

**Comment by jmklix** [comment]
Have you tried reducing the `connectTimeoutMs` to less than 3000?

**Reply by mengqingmeng** [reply]
I've tried just now,reducing the `connectTimeoutMs `to 1000, upload file still cost more than 60 seconds.

**Reply by jmklix** [reply]
Are you uploading a large file? Do you have a limited connection? Are you just wanting the sdk to return a failure faster? 

Have you tried using `MultipartUpload`? or s3Crt for improved performance?
 - [s3 MultipartUpload](https://sdk.amazonaws.com/cpp/api/LATEST/aws-cpp-sdk-s3/html/class_aws_1_1_s3_1_1_model_1_1_multipart_upload.html)
 - [s3Crt MultipartUpload](https://sdk.amazonaws.com/cpp/api/LATEST/aws-cpp-sdk-s3-crt/html/class_aws_1_1_s3_crt_1_1_model_1_1_multipart_upload.html)

**Reply by jmklix** [reply, marked as answer]
Please let us know if you are still have any questions about this sdk. Minio is a third party hosing service, so this sdk isn't guaranteed to work with that.
